core/objects_test.go

```diff
@@ -92,7 +106,7 @@ func TestChallengeSanityCheck(t *testing.T) {
 				AddressesResolved: []net.IP{net.IP{127, 0, 0, 1}},
 				AddressUsed:       net.IP{127, 0, 0, 1},
 			}}
-			test.Assert(t, chall.IsSane(false), "IsSane should be true")
+			test.Assert(t, chall.IsSane(true), "IsSane should be true")
```

Comment:
Never should have been false -- compare to the test below.  The addition of the validation record simulates the completion of the challenge, so the `completed` argument to `IsSane`  should be `true`.  I admit that this is a little unrelated to the rest of the patch, but it's small.